### PR TITLE
feat: Add new volume type "SecretReferenceVolume"

### PR DESCRIPTION
**Pull request by MoritzWeber0**
The new volume can be used to mount existing secrets to a container. It's required for mounting the pure::variants license file, which was part of the k8s.py before. In this commit, it's moved to the p::v session hook.

Resolves https://github.com/DSD-DBS/capella-collab-manager/issues/938

### CODE REVIEW

**Comment by MoritzWeber0** [comment]
Tested in the staging environment with production feature model content 👍

File: backend/tests/settings/pure_variants/test_pure_variants_license_routes.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+    response = client.patch(
+        "/api/v1/settings/integrations/pure-variants",
+        json={
+            "license_server_url": "http://localhost:27000",
+        },
+    )
+
+    assert response.is_success
+    assert response.json()["license_server_url"] == "http://localhost:27000"
```
Comment (dominik003) [inline backend/tests/settings/pure_variants/test_pure_variants_license_routes.py:49]:
```suggestion
@pytest.mark.usefixtures("admin", "pure_variants_license")
def test_set_license_if_set_already(client: testclient.TestClient):
    """Update an existing license"""

    prev_license_server_url = pure_variants_license.license_server_url

    response = client.patch(
        "/api/v1/settings/integrations/pure-variants",
        json={
            "license_server_url": "http://localhost:27000",
        },
    )

    pure_variants_config = pure_variants_crud.get_pure_variants_configuration(
        db
    )

    assert pure_variants_config
    assert pure_variants_config.license_server_url == "http://localhost:27000"

    assert prev_license_server_url != pure_variants_config.license_server_url

    assert response.is_success
    assert response.json()["license_server_url"] == "http://localhost:27000"
```
For me this test here does not completely check the update process as it is.

1. I would always include some check for the precondition. In this case, I would check that the previous license server url was other than `http://localhost:27000`
2. I would include a check that the value was actually changed in the database, i.e., not only returned by the route
3. Personally, I would always either use `localhost` or `127.0.0.1` and just adjust the port, i.e., from `27000` to `28000` as such a change is easier to see for me. However, if for you the current naming works better than we can also keep it

**Review by dominik003 — DISMISSED** [review]
It's really nice that you've included so many new tests in this project. There are some changes I suggest before we merge this, but I have already included specific suggestions and/or code for most of my comments.

File: backend/capellacollab/sessions/operators/k8s.py [hunk 2]
```diff
@@ -422,12 +422,18 @@ def _map_volume_to_k8s_volume(self, volume: models.Volume):
                     claim_name=volume.volume_name
                 ),
             )
-
-        if isinstance(volume, models.EmptyVolume):
+        elif isinstance(volume, models.EmptyVolume):
```
[the remark this answers is not in the document]
Comment (MoritzWeber0) [reply backend/capellacollab/sessions/operators/k8s.py:425]: You proposal seems to work, but it's not trivial why it does as I'm used to other programming languages. By just reading the code, I'd expect that volume would match an instance of `models.PersistentVolume`. 

I changed it to use type(volume) as value for matching.

[the remark this answers is not in the document]
Comment (MoritzWeber0) [reply backend/capellacollab/sessions/operators/k8s.py]: EDIT: Changed it to your proposal since mypy can't evaluate `type(volume)`.

**Pull request merged**